daml-script/dump/src/test/scala/com/daml/script/dump/EncodeSimpleSpec.scala

```diff
@@ -33,15 +35,26 @@ class EncodeSimpleSpec extends AnyFreeSpec with Matchers {
               exerciseResult = Some(ContractId("cid3")),
               actingParties = Seq(Party("Bob")),
             ),
+            TestData.Created(ContractId("cid4"), submitters = Seq(Party("Alice"))),
```

Comment:
Can we differentiate between `createAndExercise` and `create` + `exerciseByKey`? Seems perfectly fine if we translate the latter in the former, just curious.

Reply:
I'm not aware of a way to differentiate them in general. However, we do have access to a contract's contract key, if any, in the `createdEvent`. This could help us handle a few more cases of https://github.com/digital-asset/daml/issues/9155. Here's the approach I have in mind:
- If the exercise follows right after the create, just use `createAndExerciseCmd`.
- If the exercise follows later and the contract has a contract key, then use `exerciseByKeyCmd`.

Reply:
Addressed in https://github.com/digital-asset/daml/pull/9185